### PR TITLE
chore: bump naga to a1840be

### PR DESCRIPTION
**Pull request by AaronO**
So tip no longer indirectly pins `indexmap`, etc...

### CODE REVIEW

File: wgpu-hal/src/metal/device.rs [hunk 1]
```diff
@@ -117,7 +117,10 @@ impl super::Device {
         let mut immutable_buffer_mask = 0;
         for (var_handle, var) in module.global_variables.iter() {
             if var.class == naga::StorageClass::WorkGroup {
-                let size = module.types[var.ty].inner.span(&module.constants);
+                let size = module.types[var.ty]
+                    .inner
+                    .try_size(&module.constants)
```
Comment (kvark) [inline wgpu-hal/src/metal/device.rs:122]: should just call `size()` here

**Review by kvark — COMMENTED** [review]
One small thing

File: wgpu-hal/src/metal/device.rs (hunk 1 above)
Comment (AaronO) [reply wgpu-hal/src/metal/device.rs]: Fixed

**Review by kvark — APPROVED** [review]

File: wgpu-hal/src/gles/device.rs [hunk 2]
```diff
@@ -858,6 +858,7 @@ impl crate::Device<super::Api> for super::Device {
                 version: self.shared.shading_language_version,
                 writer_flags,
                 binding_map,
+                push_constant_binding: 0, //TODO?
```
Comment (kvark) [inline wgpu-hal/src/gles/device.rs:861]: Yes, I think doing this should work. The backend still isn't exposing PUSH_CONSTANTS feature, so it should fail the validation before using this `0`.

**Review by kvark — APPROVED** [review]

**Pull request merged**